components/brave_shields/browser/referrer_whitelist_service.cc

```diff
@@ -37,11 +37,24 @@ ReferrerWhitelistService::ReferrerWhitelist::ReferrerWhitelist(
 ReferrerWhitelistService::ReferrerWhitelist::~ReferrerWhitelist() = default;
 
 bool ReferrerWhitelistService::IsWhitelisted(
-    const GURL& firstPartyOrigin, const GURL& subresourceUrl) const {
-  for (auto rw : referrer_whitelist_) {
-    if (rw.first_party_pattern.MatchesURL(firstPartyOrigin)) {
+    const GURL& first_party_origin, const GURL& subresource_url) const {
+  if (BrowserThread::CurrentlyOn(BrowserThread::IO)) {
+    return IsWhitelisted(
+        referrer_whitelist_io_thread_, first_party_origin, subresource_url);
+  } else {
+    return IsWhitelisted(
+        referrer_whitelist_, first_party_origin, subresource_url);
+  }
+}
+
+bool ReferrerWhitelistService::IsWhitelisted(
+    const std::vector<ReferrerWhitelist> whitelist,
```

Comment:
should be `const std::vector<ReferrerWhitelist>& whitelist`

Reply:
good catch, fixed